crates/topos/src/components/setup/subnet/installer.rs

```diff
@@ -41,30 +68,35 @@ async fn download_binary(file_name: &str, uri: &str, target_directory: &Path) ->
 
     let response = reqwest::get(uri).await.map_err(Error::Http)?;
     let download_file_path = target_directory.join(Path::new(file_name));
-    let mut target_file = match File::create(&download_file_path) {
-        Err(e) => {
-            error!("Unable to create file: {e}");
-            return Err(Error::File(e));
-        }
-        Ok(file) => file,
-    };
+    {
+        //Download file
+        let mut target_archive_file = match File::create(&download_file_path) {
+            Err(e) => {
+                error!("Unable to create file: {e}");
+                return Err(Error::File(e));
+            }
+            Ok(file) => file,
+        };
```

Comment:
You could use something like `map_err` for this part

```suggestion
        let mut target_archive_file = File::create(&download_file_path).map_err(|error| {
                error!("Unable to create file: {error}");
                Error::File(error)
            })?;
```